src/prng/hc128.rs

```diff
@@ -40,14 +40,14 @@ const SEED_WORDS: usize = 8; // 128 bit key followed by 128 bit iv
 /// current state of known attacks / weaknesses of HC-128 is given in [4].
 ///
 /// The average cycle length is expected to be
-/// 2<sup>1024*32-1</sup> = 2<sup>32767</sup>.
+/// 2<sup>1024*32+10-1</sup> = 2<sup>32777</sup>.
```

Comment:
Which counter — we have a 10-bit counter?

Wait, why is `counter1024` incremented with `wrapping_add` when its type is `usize`?

Reply:
Yes, only the last 10 bits of the counter are of interest to the algorithm. We could mask out the rest, but I thought keeping track of the number of performed rounds doesn't hurt. `wrapping_add` is for when the counter does wrap around on 32-bit.